examples/plugins/tests/magic/avocado_magic/resolver.py

```diff
@@ -36,6 +36,10 @@ def resolve(reference):
                 reference,
                 ReferenceResolutionResult.NOTFOUND,
                 info='Word "%s" is not a valid magic word' % (reference))
+        if reference is EmptyReference:
```

Comment:
I can confirm the point in my previous comment:

```
[cleber@localhost magic]$ avocado -V list --resolver 
Type Test Tag(s)

Resolver             Reference                                      Info
...
magic                <class 'avocado.core.resolver.EmptyReference'> Word "<class 'avocado.core.resolver.EmptyReference'>" is not a valid magic word
...
```

Also, it looks like we can do a better job in the other resolvers too:

```
avocado-instrumented <class 'avocado.core.resolver.EmptyReference'> stat: path should be string, bytes, os.PathLike or integer, not type
exec-test            <class 'avocado.core.resolver.EmptyReference'> stat: path should be string, bytes, os.PathLike or integer, not type
golang               <class 'avocado.core.resolver.EmptyReference'>
```

Ideally, it wouldn't require boilerplate and repetitive code on all the resolvers... /me thinking about it.

Reply:
I didn't test it with `list` option. I will have to come up with a different solution.  Maybe it's possible to create different resolver method for dealing with empty reference instead of `resolve`.

Reply:
You may be on the right track there... I guess something like `get_resolutions()` (or a better name) may be a nice complementary (optional?) interface to resolvers.